users/migrations/0006_user_global_id.py

```diff
@@ -1,22 +1,31 @@
 # Generated by Django 4.2.19 on 2025-03-11 12:53
 
+import logging
+
 from django.db import migrations, models
 
+log = logging.getLogger(__name__)
+
+BATCH_SIZE = 10_000
+
 
 def transfer_global_id(apps, schema_editor):
     """Transfer the social uid to the user global_id field"""
     UserSocialAuth = apps.get_model("social_django", "UserSocialAuth")
     User = apps.get_model("users", "User")
 
-    users_with_uid = User.objects.annotate(
+    users_with_uid = User.objects.filter(
+        id__in=User.objects.filter(global_id__isnull=True).only("id")[:BATCH_SIZE]
```

Comment:
Needs an additional `global_id__isnull=False` to avoid an infinite loop - using the `while` syntax depends on the `update()` call updating the rows for a given batch so that they won't match on the next call since you're always fetching the first 10k rows.